File: PhaseGateJava/Mbti.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;
public class Mbti{

	
	static List<List<String>> allQuestions = new ArrayList<>();

	public static List<List<String>> setOne(){
        	List<String> extrovertedEVsIntrovertedI = new ArrayList<>();

        	extrovertedEVsIntrovertedI.add("expand energy, enjoy groups");
        	extrovertedEVsIntrovertedI.add("conserve energy, enjoy one-on-one");
        	extrovertedEVsIntrovertedI.add("more outgoing, think out loud");
        	extrovertedEVsIntrovertedI.add("more reserved, think to yourself");
        	extrovertedEVsIntrovertedI.add("seek many tasks, public activities, interaction with others");
        	extrovertedEVsIntrovertedI.add("seek private, solitary activities with quiet to concentrate");
        	extrovertedEVsIntrovertedI.add("external, communicative, express yourself");
        	extrovertedEVsIntrovertedI.add("internal, reticent, keep to yourself");
        	extrovertedEVsIntrovertedI.add("active, initiate");
        	extrovertedEVsIntrovertedI.add("reflective, deliberate");
        	allQuestions.add(extrovertedEVsIntrovertedI);
        	return allQuestions;
    	}

	public static List<List<String>> setTwo(){        
		List<String> sensingSVsIntuitiveN = new ArrayList<>();

		sensingSVsIntuitiveN.add("interpret literally");
		sensingSVsIntuitiveN.add("look for meaning and possibilities");
		sensingSVsIntuitiveN.add("practical, realistic, experimental");
		sensingSVsIntuitiveN.add("imaginative, innovative, theoretical");
		sensingSVsIntuitiveN.add("standard, usual, conventional");
		sensingSVsIntuitiveN.add("different, novel, unique");
		sensingSVsIntuitiveN.add("focus on here-and-now");
		sensingSVsIntuitiveN.add("look to the future, global perspective, big picture");
		sensingSVsIntuitiveN.add("facts, things, what is");
		sensingSVsIntuitiveN.add("ideas, dreams, what could be, philosophical");
		allQuestions.add(sensingSVsIntuitiveN);
		return allQuestions;
	}

	public static List<List<String>> setThree(){
		List<String> thinkingTVsFeelingF = new ArrayList<>();

 		thinkingTVsFeelingF.add("logical, thinking, questioning");
		thinkingTVsFeelingF.add("empathetic, feeling, accommodating");
		thinkingTVsFeelingF.add("candid, straightforward, frank");
		thinkingTVsFeelingF.add("tactful, kind, encouraging");
		thinkingTVsFeelingF.add("firm, tend to criticize, hold the line");
		thinkingTVsFeelingF.add("gentle, tend to appreciate, conciliate");
		thinkingTVsFeelingF.add("tough-minded, just");
		thinkingTVsFeelingF.add("tender-hearted, merciful");
		thinkingTVsFeelingF.add("matter of fact, issue-oriented");
		thinkingTVsFeelingF.add("sensitive, people-oriented, compassionate");
		allQuestions.add(thinkingTVsFeelingF);
		return allQuestions;
	}

	public static List<List<String>> setFour(){
		List<String> judgingJVsPerceptionP = new ArrayList<>();
		judgingJVsPerceptionP.add("organized, orderly");
		judgingJVsPerceptionP.add("flexible, adaptable");
		judgingJVsPerceptionP.add("plan, schedule");
		judgingJVsPerceptionP.add("unplanned, spontaneous");
		judgingJVsPerceptionP.add("regulated, structured");
		judgingJVsPerceptionP.add("easy-going, live and let live");
		judgingJVsPerceptionP.add("preparation, plan ahead");
		judgingJVsPerceptionP.add("go with the flow, adapt as you go"); 
		judgingJVsPerceptionP.add("control, govern");
		judgingJVsPerceptionP.add("latitude, freedom");
		allQuestions.add(judgingJVsPerceptionP);
		return allQuestions;
	}
	
	public static void printAllQuestions(){
	int aCount = 0;
	int bCount = 0;
	Scanner scanner = new Scanner(System.in);
		for(int count = 0; count < 10; count+=2){
			for(int counter = 0; counter < allQuestions.size() ; counter++){
				List<String> currentQuestion = allQuestions.get(counter);
				System.out.print(" " + currentQuestion.get(count) + " ");
				System.out.println(" " + currentQuestion.get(count + 1) + " ");
				String answer = scanner.nextLine().toLowerCase();
				if(answer.equals("a")){
					aCount++;
				}
				else if(answer.equals("b")){
					bCount++;
				}
				else{
				break;
				
				}
			
			}
			System.out.println();
		}
		
	}		

	

	
	
}

File: phase_gate_py/mbti.py
all_questions = []

def set_one():
    extroverted_vs_introverted = [
        "expand energy, enjoy groups",
        "conserve energy, enjoy one-on-one",
        "more outgoing, think out loud",
        "more reserved, think to yourself",
        "seek many tasks, public activities, interaction with others",
        "seek private, solitary activities with quiet to concentrate",
        "external, communicative, express yourself",
        "internal, reticent, keep to yourself",
        "active, initiate",
        "reflective, deliberate"
    ]
    all_questions.append(extroverted_vs_introverted)
    return all_questions

def set_two():
    sensing_vs_intuitive = [
        "interpret literally",
        "look for meaning and possibilities",
        "practical, realistic, experimental",
        "imaginative, innovative, theoretical",
        "standard, usual, conventional",
        "different, novel, unique",
        "focus on here-and-now",
        "look to the future, global perspective, big picture",
        "facts, things, what is",
        "ideas, dreams, what could be, philosophical"
    ]
    all_questions.append(sensing_vs_intuitive)
    return all_questions

def set_three():
    thinking_vs_feeling = [
        "logical, thinking, questioning",
        "empathetic, feeling, accommodating",
        "candid, straightforward, frank",
        "tactful, kind, encouraging",
        "firm, tend to criticize, hold the line",
        "gentle, tend to appreciate, conciliate",
        "tough-minded, just",
        "tender-hearted, merciful",
        "matter of fact, issue-oriented",
        "sensitive, people-oriented, compassionate"
    ]
    all_questions.append(thinking_vs_feeling)
    return all_questions

def set_four():
    judging_vs_perceiving = [
        "organized, orderly",
        "flexible, adaptable",
        "plan, schedule",
        "unplanned, spontaneous",
        "regulated, structured",
        "easy-going, live and let live",
        "preparation, plan ahead",
        "go with the flow, adapt as you go",
        "control, govern",
        "latitude, freedom"
    ]
    all_questions.append(judging_vs_perceiving)

def print_all_questions():
    a_count = 0
    b_count = 0
    print("Answer with 'a' or 'b':\n")

    for i in range(0, 10, 2):  # Go through pairs (0&1, 2&3, ...)
        for category in all_questions:
            print(f"A: {category[i]}  |  B: {category[i + 1]}")
            answer = input("Your choice (a/b): ").strip().lower()
            if answer == 'a':
                a_count += 1
            elif answer == 'b':
                b_count += 1
            else:
                print("Invalid input, skipping question.\n")
        print()  # Newline between rounds

    print(f"Total A answers: {a_count}")
    print(f"Total B answers: {b_count}")
